plugin/db/mysql/rollback.go

```diff
@@ -0,0 +1,273 @@
+package mysql
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	regexpDatabaseTable = regexp.MustCompile("`(.+)`\\.`(.+)`")
+)
+
+// GetRollbackSQL generates the rollback SQL for the list of binlog events in the reversed order.
+func (txn BinlogTransaction) GetRollbackSQL(columnNames []string) (string, error) {
+	if len(txn) == 0 {
+		return "", nil
+	}
+	var sb strings.Builder
+	_, _ = sb.WriteString("BEGIN;\n")
+	// Generate rollback SQL for each statement of the transaction in reversed order.
+	// Each statement may have multiple affected rows in a single binlog event. The order between them is irrelevant.
+	for i := len(txn) - 1; i >= 0; i-- {
+		e := txn[i]
+		if e.Type != WriteRowsEventType && e.Type != DeleteRowsEventType && e.Type != UpdateRowsEventType {
+			continue
+		}
+		sql, err := e.getRollbackSQL(columnNames)
+		if err != nil {
+			return "", err
+		}
+		_, _ = sb.WriteString(sql)
```

Comment:
Below we use strings.Join(sqlList, "\n"), but we use different method here. Let's be consistent.